boards/posix/native_posix/cmdline.c

```diff
@@ -8,12 +8,12 @@
 #include <stdlib.h>
 #include <string.h>
 #include "cmdline_common.h"
-#include "zephyr/types.h"
+#include <zephyr/types.h>
 #include "hw_models_top.h"
 #include "timer_model.h"
 #include "cmdline.h"
-#include "toolchain.h"
-#include "posix_trace.h"
+#include <toolchain.h>
+#include <arch/posix/posix_trace.h>
```

Comment:
This header is actually meant to be fully public.
(Hiding it, breaking compatibility with apps or components off-tree is not ok.)

Reply:
This file contained "posix arch"-only functions and that's why I placed it under include/arch/posix/ (i.e. if you call these public functions from anything other than "posix arch", they will not work).

Do you still want me to place it directly under include/? (it was previously inside arch/posix/include/).

Alternatively, I think it might make more sense to include it from include/arch/posix/arch.h, so that it is not necessary to manually include that file using that rather long path (after all, since posix is something intended for testing only, you'd almost always want tracing).

Let me know what you think.

Reply:
I would not place it in the top include/ :),
it can be anywhere where it can included as before this PR if `CONFIG_ARCH_POSIX` is selected. (meaning all changes where you needed to do
`#include "posix_trace.h" => #include <arch/posix/posix_trace.h>` should not be necessary )

Reply:
> it can be anywhere where it can included as before this PR if `CONFIG_ARCH_POSIX` is selected.

With arch/*/include/ and kernel/include/ gone from the public include paths, that leaves me with:
```
include/
include/drivers/
soc/posix/inf_clock/
soc/posix/inf_clock/include
```

include/ - is probably not the best place, as we both already agreed on.
include/drivers - well, you could think of it as a "posix trace driver" ..., but one could easily argue this is, strictly speaking, not a driver like the others.
soc/posix/inf_clock - if we declare that there will only ever be "inf_clock SoC" for posix, this could be a potential candidate.

Since you are the author of this port, I will follow your decision.

Reply:
> soc/posix/inf_clock/ - if we declare that there will only ever be "inf_clock SoC" for posix

Nope.

What about a new `arch/posix/include/public/` ?

Reply:
> What about a new arch/posix/include/public/ ?

I don't really see the point of adding another arch-specific public directory when there is already include/arch/*/.

@andrewboie what do you think about this? 

@aescolar Is there a specific reason as to why posix_trace.h has to be selectively included by the end users? Couldn't they be, for example, just implied by include/arch/posix/arch.h (hence by kernel.h and all files that include this)?

Reply:
> I don't really see the point of adding another arch-specific public directory when there is already include/arch/*/.

I don't either.
If it's public, place it in include/arch/posix/

Reply:
Yep, that is fine , for some reason I though that directory was not there anymore after this PR while answering before

Reply:
@aescolar If we place posix_trace.h in include/arch/posix, we will have to do either:

1. `#include <arch/posix/posix_trace.h>` as this PR already does OR
2. include posix_trace.h from include/arch/posix/arch.h so that it is not necessary to manually include posix_trace.h from other places.

Adding include/arch/${ARCH} to global include path will cause a lot of issues as there are many conflicting file names in that directory (for example, arm will definitely have a problem with that).

Since you don't seem to like the idea of the first approach, should I go with the second one then?

Reply:
@stephanosio Any posibiltiy of doing it without affecting all users of that header? Your second proposal will still require all users to remove those include lines, correct?

Reply:
> Your second proposal will still require all users to remove those include lines, correct?

@aescolar Correct. It would be less ugly than adding `#include <arch/posix/posix_trace.h>` everywhere though.

> Any posibiltiy of doing it without affecting all users of that header?

Deprecation would be one way to do it. We can `#include <arch/posix/posix_trace.h>` from include/arch/posix/arch.h so that all the existing third-party code will still function as is. In the meantime, we can temporarily add posix_trace.h under include/ with `#warning "posix_trace.h is deprecated and will be removed in a future release. Please include arch/cpu.h (already included by kernel.h and zephyr.h) instead."`, so that users will have some time to remove `#include <posix_trace.h>` in the application code until this header file gets completely removed in the future.

Reply:
>  In the meantime, we can temporarily add posix_trace.h under include/ with `#warning `

sounds like a good idea.

Reply:
@aescolar For now, I pushed a new commit with the deprecation strategy, while fixing other unrelated merge conflicts.

Let me know if this is acceptable.